tests/test_griddeddata.py

```diff
@@ -98,8 +99,19 @@ def test_GriddedData_longitude(data_tm5):
     assert str(data_tm5.longitude.units) == "degrees"
 
     lons = data_tm5.longitude.points
-    assert_allclose(lons.min(), -181.5, rtol=TEST_RTOL)
-    assert_allclose(lons.max(), 175.5, rtol=TEST_RTOL)
+    # iris >= 3.2 corrected an error in iris.cube.Cube.intersection
+    # this does not work with iris:
+    # from importlib.metadata import version
+    # import iris
+    # version('iris')
+    # using iris.__version__ instead
+    version = iris.__version__.split(".")
+    if int(version[0]) >= 3 and int(version[1]) > 1:
+        assert_allclose(lons.min(), -178.5, rtol=TEST_RTOL)
+        assert_allclose(lons.max(), 178.5, rtol=TEST_RTOL)
+    else:
+        assert_allclose(lons.min(), -181.5, rtol=TEST_RTOL)
+        assert_allclose(lons.max(), 175.5, rtol=TEST_RTOL)
```

Comment:
I would allocate the expected values on the if clause and call `assert_allclose`
after. Like this:

``` python
# the results depend on the version of iris
# https://github.com/metno/pyaerocom/issues/588
iris_version = tuple(map(int, iris.__version__.split(".")))
if iris_version > (3, 1):
    lon_min, lon_max = -178.5, 178.5 # correct values
else:
    lon_min, lon_max = -181.5, 175.5
assert_allclose(lons.min(), lon_min, rtol=TEST_RTOL)
assert_allclose(lons.max(), lon_max, rtol=TEST_RTOL)    
```

Reply:
also could you check that the `assert_allclose` are really needed?
Maybe it would be enough with
``` Python
assert lons.min(), lons.max() == lon_min, lon_max
```